coil-core/src/commonMain/kotlin/coil3/ImageLoader.kt

```diff
@@ -213,6 +218,32 @@ interface ImageLoader {
             )
         }
 
+        /**
+         * The [CoroutineContext] that any UI/main thread work will be executed in.
+         *
+         * NOTE: This should typically only be used in tests, as `View`s and `Lifecycle`s will be
+         * accessed in this context and those operations will throw if not performed on a valid
+         * thread.
+         *
+         * Default: `Dispatchers.Main.immediate`
+         */
+        @ExperimentalCoilApi
+        fun mainCoroutineContext(context: CoroutineContext) = mainCoroutineContext { context }
+
+        /**
+         * The [CoroutineContext] that any UI/main thread work will be executed in.
+         *
+         * NOTE: This should typically only be used in tests, as `View`s and `Lifecycle`s will be
+         * accessed in this context and those operations will throw if not performed on a valid
+         * thread.
+         *
+         * Default: `Dispatchers.Main.immediate`
+         */
+        @ExperimentalCoilApi
+        fun mainCoroutineContext(initializer: () -> CoroutineContext) = apply {
```

Comment:
We need to have a lazy method as `Dispatchers.main.immediate` can throw on desktop in certain (expected) cases.